ui/js/controllers/perf/compare.js

```diff
@@ -525,6 +462,57 @@ perf.controller('CompareSubtestResultsCtrl', [
             $scope.$apply();
         }
 
+        async function fetchData() {
+            const createQueryParams = (parent_signature, repository) => ({
+                parent_signature,
+                framework: $scope.filterOptions.framework,
+                repository,
+            });
+            
+            const originalParams = createQueryParams($scope.originalSignature, $scope.originalProject.name);
+            let results;
+            let newResults;
+            let originalResults;
+
+            if ($scope.originalRevision) {
+                originalParams.revision = $scope.originalResultSet.revision;
+            } else {
+                // TODO create a helper for the startday and endday since this is also used in compare view
+                const startDateMs = ($scope.newResultSet.push_timestamp -
+                    $scope.selectedTimeRange.value) * 1000;
+                const endDateMs = $scope.newResultSet.push_timestamp * 1000;
+
+                originalParams.startday = new Date(startDateMs).toISOString().slice(0, -5);
+                originalParams.endday = new Date(endDateMs).toISOString().slice(0, -5);
+            }
+
+            // Is there ever a use case where originalSignature is provided but newSignature isn't?
```

Comment:
I can't personally think of one... this might be an error case?

I think this code could probably use some additional refactoring beyond the scope of this PR-- it's still a bit of a confusing nest of if...else...if...else which we could probably clean up by more cleanly defining its inputs/outputs.

Reply:
yeah, I plan to clean it up a bit more once I convert these views to components.

Reply:
Removed this conditional and tested it.